bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/AbstractComponent.java

```diff
@@ -98,8 +130,14 @@ public AbstractComponent(CFactory.ComponentConfiguration componentConfiguration,
      */
     public CompletableFuture<@Nullable Void> start(MqttBrokerConnection connection, ScheduledExecutorService scheduler,
             int timeout) {
-        return channels.values().stream().map(v -> v.start(connection, scheduler, timeout))
-                .reduce(CompletableFuture.completedFuture(null), (f, v) -> f.thenCompose(b -> v));
+        CompletableFuture<@Nullable Void> all = CompletableFuture.completedFuture(null);
+
+        if (availablityChannel != null) {
+            all = all.thenCompose(v -> availablityChannel.start(connection, scheduler, timeout));
+        }
+
+        return channels.values().stream().map(v -> v.start(connection, scheduler, timeout)).reduce(all,
```

Comment:
Not 100% sure, but I think this might be wrong. The first parameter of `reduce` should be the identity element (i.e. `CompletableFuture.completedFuture(null)`, after your changes it is not. Can't you just add the `availabilityChannel` to `channels` (if available)?

Reply:
I an confident, that this works, but you are right, that it does not 100% comply with the specs.
I can change the order (first the stream->reduce and then add the <code>availabilityChannel</code>
I did not want to add the <code>availabilityChannel</code> to <code>channels</code>, because then it would show up in the front-end. Which i would not like.
I would want to use the channel to control the thing state only.

Reply:
Good point.